homeassistant/components/husqvarna_automower/sensor.py

```diff
@@ -0,0 +1,173 @@
+"""Creates a the sensor entities for the mower."""
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from aioautomower.model import MowerAttributes
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory, UnitOfLength, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AutomowerDataUpdateCoordinator
+from .entity import AutomowerBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class AutomowerSensorEntityDescription(SensorEntityDescription):
+    """Describes WLED sensor entity."""
+
+    exists_fn: Callable[[MowerAttributes], bool] = lambda _: True
+    value_fn: Callable[[MowerAttributes], str]
+
+
+SENSOR_TYPES: tuple[AutomowerSensorEntityDescription, ...] = (
+    AutomowerSensorEntityDescription(
+        key="battery_percent",
+        translation_key="battery_percent",
```

Comment:
Can be removed. If no translation key is found and no name is provided it will fallback to the device class translations.

Reply:
done in https://github.com/home-assistant/core/pull/110410/commits/63e53a206dc05a5fa27423df05ac7dedaf4b4ab2